### PR TITLE
chore(ci): add cypress to chck start script

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Types of changes

Launch slicemachine in github actions. 

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

We want to check the start script in the ci, this PR aims to do that, and also adds cypress :)

This branch will have to be merged onto or after https://github.com/prismicio/slice-machine/pull/166

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/lib/models/common/CustomType/tab.ts [hunk 1]
```diff
@@ -175,34 +175,35 @@ export const Tab = {
...
-          if (curr.value.type === FieldType.Group) {
-            return {
-              ...acc,
-              groups: [
-                ...acc.groups,
-                { key: curr.key, value: curr.value as GroupField<AsArray> },
-              ],
-            };
-          }
+    const { fields, groups }: OrganisedFields = tabAsArray.value.reduce<
```
Comment (hypervillain) [inline packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: was this EOL created by our linter?

File: packages/slice-machine/lib/models/common/http/FakeClient.ts [hunk 2]
```diff
@@ -8,7 +8,9 @@ export interface FakeResponse {
 }
 
 export default class FakeClient {
-  private mutate: (jsonResponse?: any) => {
+  private mutate: (
+    jsonResponse?: any
```
Comment (hypervillain) [inline packages/slice-machine/lib/models/common/http/FakeClient.ts:12]: same question here, I don't understand the rule that's applied here

File: packages/slice-machine/lib/models/common/CustomType/tab.ts (hunk 1 above)
Comment (hypervillain) [inline packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: Was this EOL created by our linter?

File: packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
+/// <reference types="cypress" />
```
Comment (hypervillain) [inline packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js]: I believe we can get rid of these examples

File: packages/slice-machine/tests/project/cypress/plugins/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="cypress" />
```
Comment (hypervillain) [inline packages/slice-machine/tests/project/cypress/plugins/index.js:1]: same for these files, I don't think they make sense here

**Review by hypervillain — CHANGES_REQUESTED** [review]

File: packages/slice-machine/lib/models/common/CustomType/tab.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: Yes

Comment (MarcMcIntosh) [reply packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: There are a few other issues picked up by prettier,
I think one of them was a weird looking type definition.
```
packages/slice-machine/lib/models/ui/LibraryState.ts
[error] packages/slice-machine/lib/models/ui/LibraryState.ts: SyntaxError: ',' expected. (7:35)
[error]   5 |   name: string;
[error]   6 |   isLocal: boolean;
[error] > 7 |   components: ReadonlyArray<[state: SliceState, store: Store]>;
[error]     |                                   ^
[error]   8 | }
[error]   9 |
```

File: packages/slice-machine/lib/models/common/http/FakeClient.ts (hunk 2 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/models/common/http/FakeClient.ts:12]: it's not a linting rule, it's prettier.... could have something to do with line length or method declarations (i'm not sure tbh)

File: packages/slice-machine/lib/models/common/CustomType/tab.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: yes

File: packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js]: Yeah I figured keep them would be good for future reference. they don't get run but do provided some documentation on how to use cypress, wdyt?

File: packages/slice-machine/tests/project/cypress/plugins/index.js (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/tests/project/cypress/plugins/index.js:1]: Plugins?, 
We'll probably end-up needing them at some point.

File: packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js]: I've removed the examples

Comment (hypervillain) [reply packages/slice-machine/tests/project/cypress/examples/1-getting-started/todo.spec.js]: Sorry I missed your comment! I get your point too, but I thought it wouldn't be too difficult to start from the Cypress documentation

File: packages/slice-machine/lib/models/common/CustomType/tab.ts (hunk 1 above)
Comment (hypervillain) [reply packages/slice-machine/lib/models/common/CustomType/tab.ts:178]: hmm yes I see why it complained. Some EOL seem a bit weird to me but let's see how it goes and it's not a big deal if 99% of the rest is better now

**Comment by hypervillain** [comment]
The tests failed because of a bad test of mine. @Miniplop I think you have something in the pipe that fixes it right?

**Review by Miniplop — APPROVED** [review]

File: packages/slice-machine/package.json [hunk 5]
```diff
@@ -26,7 +26,7 @@
     "slicemachine": "start-slicemachine --port 9999",
     "security-audit:prod": "yarn audit --production --only=prod --audit-level=low",
     "security-audit:dev": "yarn audit --only=dev --audit-level=moderate",
-    "lint": "eslint --max-warnings 3186 --ext .js,.jsx,.ts,.tsx .",
+    "lint": "eslint --max-warnings 3280 --ext .js,.jsx,.ts,.tsx .",
```
Comment (Heyrows) [inline packages/slice-machine/package.json:29]: this number should decrease not inscrease lol

File: packages/slice-machine/tests/project/cypress/integration/0-is-running/check.spec.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/// <reference types="cypress" />
+
+describe("check slicemachine is running", () => {
+  it('visit http://localhost:9999', () => {
+    cy.visit('http://localhost:9999')
```
Comment (Heyrows) [inline packages/slice-machine/tests/project/cypress/integration/0-is-running/check.spec.js:5]: actually found this site that does what we want, this should be cy.visit('/') with the url as environment variable.

Comment (Heyrows) [reply packages/slice-machine/tests/project/cypress/integration/0-is-running/check.spec.js:5]: https://docs.cypress.io/guides/getting-started/testing-your-app#Step-3-Configure-Cypress

**Pull request merged**